### ISSUE TITLE
AT-Missile of Titan-Launcher self-explodes in the air, before it hits the target!

### ISSUE THREAD

**Issue by ZeusWrath** [acemod/ACE3#7267]
**Mods (complete and add to the following information):**
- **Arma 3:** `1.96` [stable]
- **CBA:** `3.12.1.190812` [ stable]
- **ACE3:** `3.12.6` [ stable]

**Description:**
Hi ACE3!
 
Sometimes, when player firing a Titan Launcher Compact, an ACE3 user observes a missile not reaching its target and exploding in the air for no apparent reason. 

I made video - https://www.youtube.com/watch?v=hYGHaXc0_gk&feature=youtu.be

This happens both with the player and with the AI ​​who will shoot from the Titan Launcher Compact. Typically, this occurs at large distances to the target.

Undisputed Logic here:

1) If the shot was fired, then the target was locked. Without a target capture, the Titan is unable to shoot at all! I mean AT-missile.
2) If the target was locked, then the AT-missile knows where to fly and this missile should reach the target at any distance and under any circumstances, if there are no obstacles on the way to the target. Is it logical? I think yes.

But instead, an ACE3 user periodically watches missile explode in the air one by one, as if it were a festive firework! AI shoots one missile in a row! One, two, three - And all of this missiles, one by one, explode in the air!  LOL - such a sight is not for Arma3! 

Dear developers! Please correct this BUG.  I don’t know what this is connected with, but I know that you use your system for Guided Missiles in Your mod. It seems to me like an artificial limitation of the distance to the target. It’s as if the distance limit makes the AT-missile explode, but it’s not right!  If the limit should exist, then it should be on _ability of capturing the target_, but not on the flight of the AT-missile to this target.

**Steps to reproduce:**
Play with AT-Soldiers and enemy vehicles at med and big distnces

**Expected behavior:**
If Titan Launcher has _**locked the target**_, then the AT-Missile should follow toward it, regardless of any distance.  This should not explode on its own in the air! 
If there is a distance limit for hitting vehicles from the Titan Launcher, then the target _**should not be locked**_ in general. 

**Where did the issue occur?**
-  Singleplayer / Editor (Singleplayer) / Editor (Multiplayer) / Virtual Arsenal

**Comment by commy2** [comment]
>Undisputed Logic here

**Comment by ZeusWrath** [comment]
Hi guys!

> Why not ?

How why?  Because the target is locked! Why are you asking about this? I was sure that it was clear even to a teenager -  **If the target is locked using modern laser target capture systems, then the this missile knows where to fly.  Logically or not? This missile will fly where necessary and will not explode for no reason in the air!** Even if a missile loses its target due to some reason (electronic interference), then it should not explode, it just has to fly on.

> Undisputed Logic here
have any doubts?

> How far is the target ?
I can’t say exactly what the distance is there, sometimes I don’t understand what kind of target the AI is shooting at, but I see waveworks from explosions in the sky! I don’t understand why the AI and the player should spend their ammunition by shooting an empty piece of the sky? This is not normal, I was sure that any normal person would agree with this. If I shoot, then it seems to me that the AT-missile could explode, pretty close, around 700-1000 meters. It looks like an absurdity.

> Missiles have range limits after which they'll explode

Are you sure about this? Even if this is so, then using mods we can increase this distance. If this is so, then I do not understand why such stupidity exists in ArmA3. If need a limit for use AT-missile, then it's enough to create a limit on target lock, instead of exploding ammunition in vain.

**ACE team please do something! 
How tired I am of seeing these nonsense in Arma3. 
It looks like a shame**

**Comment by Blutze** [comment]
It would absolutely be possible for the guidance system's range to far exceed that of the rocket motor, without the system being able to measure the distance to the target at launch. These selfdestruct mechanism however only have to track the time of flight, and disable the missile just before it runs out of energy (the fuel itself could be long gone by that point, with the missile just coasting on for most of the distance).

The reason that has to happen is safety. You don't want the thing to just impact the ground halfway to the target, potentially in the middle of some civilian infrastructure.

> I can’t say exactly what the distance is there

Then find out. If you don't put any effort into providing us with as much information as possible, we can't help you.

Not to mention that I can lock onto targets 2.5 klicks away without the AT missile selfdestructing, and the SACLOS guided AP variant quits at >2.6-2.8 klicks. Meaning something is borked on your side, and you should have included an RPT in the first place.

**Comment by commy2** [comment]
Yeah, this defo needs an RPT. Who knows what other addons are running that are ruining the missile config by bad inheritance etc.

**Comment by ZeusWrath** [comment]
> It would absolutely be possible for the guidance system's range to far exceed that of the rocket motor, without the system being able to measure the distance to the target at launch. These selfdestruct mechanism however only have to track the time of flight, and disable the missile just before it runs out of energy (the fuel itself could be long gone by that point, with the missile just coasting on for most of the distance).

> The reason that has to happen is safety. You don't want the thing to just impact the ground halfway to the target, potentially in the middle of some civilian infrastructure.

You want to tell me that in real life, Javelin AT-Missile works the same way? 
Should it explode on its own?  I think this is not so

> Then find out. If you don't put any effort into providing us with as much information as possible, we can't help you. 

 I’m ready to make many efforts to solve this issue, but it seems to me that this makes no sense.  If the this limitation exists in the Arma3, then the person creating the modification of this AT-system should know about this limitation, where it is located, how to change it, and so on. If I'm wrong, then correct me.

Nevertheless, I will try to test this distance and let you know more precisely, a later.

**Comment by ZeusWrath** [comment]
> Yeah, this defo needs an RPT. Who knows what other addons are running that are ruining the missile config by bad inheritance etc.

Ok i will provide this later

**Comment by BaerMitUmlaut** [comment]
Air to air and ground to air missiles often have self destruct mechanisms. The RPG-7 also has a self destruct mechanism, which is apparently sometimes even used as an airburst mechanism by insurgents.

I'm not sure if the Javelin has a similar mechanism, the manual doesn't mention one (doesn't mean there isn't one). The flight motor is burned out completely after around 5s, with barely any power left at around 4s. So yes, the guidance time far exceeds the time the motor is on.

**Comment by alganthe** [comment]
The titan AT missile is modeled after the Rafael spike-SR not a javelin variant.

**Comment by commy2** [comment]
Titan missiles all have 15 seconds ttl after that they explode.
```cpp
    class M_Titan_AA: MissileBase {
        timeToLive = 15;
```
This is from the base game and I don't see it changed anywhere in ACE.

**Comment by ZeusWrath** [comment]
Guys! I had a distance test and I got a strange result! I shot a tank with a subordinate AI at distances of 1 km, 1.5 km, 2 km. And not once have I seen a missile explode! What the heck! I constantly see this in the game!. What conclusion can I draw?
Probably the reason for the self-explosion of a AT-Missile is not the distance to the target, or maybe not only distance!?

I can assume - the target is hiding behind the some object or hiding from sight of the missile - maybe this fact have influence!?  I don’t even know what could be the reason. But I very often see this during a fight! I will get you an RPT-file, but I need time for a test game and wash such a moment in the game.

**Comment by ZeusWrath** [comment]
Hey guys! I had a test, I saw how several AT-Missile exploded in the air before reaching the target! I put several AI subordinates on the hill and ordered to attack the tank column. The first few missiles hit the target; it was 3-4 shots. Then I saw the other 3 missile burst into the air (one by one)  If you need a video, then I will do it for you!

My RPT here - https://drive.google.com/open?id=1ReukoDh4YP9Vr_0FXkaxPXQvvKQjIM6z

I hope this helps you. Please try to fix this s*hit, it interferes with Arma3 gameplay

**Comment by ZeusWrath** [comment]
**Hey dudes, you won’t decide this! I'm tired of seeing it. AI shoots at long-range vehicles, but missiles explode in the air! It breaks the balance. Titan-Launcher gets useless as AI spends missiles in vain**

Please deal with this, somewhere in the game there is a limiter for hitting a target with Titan missiles!
AT-Missiles should not explode, if there is a restriction on the use of Titan, then the target should not be captured

**Comment by dedmen** [comment]
> Probably the reason for the self-explosion of a AT-Missile is not the distance to the target, or maybe not only distance!?

Yes. As you were told RIGHT ABOVE that message, its time, not distance.

According to wiki the Titan missiles should have a range of 2km.
It also seems to be mostly modeled after the FGM-148 Javelin, which has a flight time of 19 seconds

**Comment by bux** [comment]
Maybe he wants that AI units don't even start to fire the Titan if the range exceeds the flight time.

**Comment by ZeusWrath** [comment]
HI guys!

> Yes. As you were told RIGHT ABOVE that message, its time, not distance.
> 
> According to wiki the Titan missiles should have a range of 2km

No, No,! it is significantly less than 2 km! If the game had a 2 km limit, then I would be happy, but in fact this distance is much less. I can’t say exactly how much exactly, but the distance is < 1 km!

> I don't see anything to be done here, sure we could make the lifetime 4 seconds longer because the Javelin works for that time, but the Titan is just inspired by the javelin, its not the same, so I think the 15 seconds BI chose is reasonable.

Gameplay not should measure the lifetime of a rocket in seconds, it’s absurd! If the target has been captured by means laser and if the missile has already been launched, then this missile should reach the target and no matter how many seconds such a flight takes! Please because think it's logical!

I remember very well that the old versions of Armagh did not have such a problem, since targets at a distance of> 1.5 km could be guaranteed to be destroyed. Apparently, in some kind of update, BIS broke this. So it can be configured in the opposite direction!


> Maybe he wants that AI units don't even start to fire the Titan if the range exceeds the flight time.

Yes yes yes, and not only AI units. Let the player’s launcher not be able to capture the target, if the rocket cannot reach the target! This is logical! Otherwise, the soldier is doomed to a senseless waste of ammunition

ACE please try to do something!  There is no point in the senseless consumption of ammunition that will explode in the air

**Comment by ZeusWrath** [comment]
I need to make a demonstration video. Wait guys, I'll clarify everything now!

**Comment by ZeusWrath** [comment]
I managed to make a video!
https://www.youtube.com/watch?v=hYGHaXc0_gk&feature=youtu.be

Three missile and all three exploded before hit at distance < 1 km!!! 
This does not always happen, but I caught all three. Sometimes one missile exploded in the air, sometimes two. Most likely there is a limit on the distance.

Well, how do you still not see the problems? I think this is a complete absurdity that needs treatment.
Please cure it

**Comment by ZeusWrath** [comment]
Hey guys. I completed the question on BIS feedback - https://feedback.bistudio.com/T152723
It turned out that the vanilla AT-Titan Missile has a lifetime of 22 seconds. I used a stopwatch and saw that in my video the Missiles explode the same way after 22 seconds. 

This means that 22 is not enough. Please fix it!

You can do:

a) increase the lifetime of the AT-Titan Missile, it is better to make it very large (even endless!)
b) change the speed of the AT-Titan Missile, so that it reaches the goal faster
c) change the flight path of AT-Titan Missile, so that the flight to the target takes less time

Any solutions so that the Missile does not explode in front of the target, because it is bullshit.

**Comment by commy2** [comment]
But I told you this half a year ago already:

https://github.com/acemod/ACE3/issues/7267#issuecomment-551408710

What the TTL is ought to be is not something I want to say, but it definitely should not be infinite, as any missile runs on finite fuel.

**Comment by tbeswick96** [comment]
@ZeusWrath How is this an ACE bug? If you want to change it you can make a simple config patch mod for your needs.

**Comment by dedmen** [comment]
> Gameplay not should measure the lifetime of a rocket in seconds, it’s absurd!

no.. actually thats not absurd. Actually thats exactly how it works.

Only thing I see changing here is to make AI know that they can't reach a target thats too far away, and not attempting to fire.
But would you know in real live? would you calculate the arc the missile takes to the target before you shoot?
Also don't know if ACE could even fix that?

**Comment by bux** [comment]
>But would you know in real live

Yeah, most definitely

**Comment by ZeusWrath** [comment]
> But I told you this half a year ago already:

what did you tell me? that it lasts 15 seconds? So it is written on your link. This is not a correct value, **in a vanilla game it lasts 22 seconds.** This was written to me by BIS on the feedback forum! It can also be calculated in my video!

> What the TTL is ought to be is not something I want to say, but it definitely should not be infinite, as any missile runs on finite fuel. 

**Maybe am I arguing with this !? You again do not understand me?  When I say infinite, I mean infinite lifetime for a missile body! It shouldn't explode in empty air,  once it runs out of fuel, it should fall to the ground and explode on the fall.**

@commy2 
@tbeswick96
@dedmen

Dudes! 

I don't understand why I should ask you to make your work adequate?  Why can't you all see that this is bullshit?  
I made a video for you - 

**three missile launches and three useless explosions in the air?** - This is normal?  Just answer me  this question. If you answer me - "YES ITS NORMAL"  then I'll close the topic.  An adequate person will not answer "yes this is normal", he will say yes there is a problem and it needs to be treated

**Comment by tbeswick96** [comment]
Yes it is normal. Missiles that are designed to explode when they reach a certain limit will do so.

You're not understanding where the perceived "error" is. The time to live value is 22 seconds as you've been told, and found in-game.
The 15 seconds quoted was probably for a different missile. The important thing to note about that comment is `This is from the base game and I don't see it changed anywhere in ACE.`

Basically, what you are seeing as an error, is A) normal, and B) not touched by ACE, therefore is not an ACE bug, and therefore does not belong on this issue tracker.

**Comment by ZeusWrath** [comment]
> Yes it is normal

_**The missile is made according to the classic aerodynamic design with folding wings. The missile of the Javelin complex is equipped with an infrared homing head (IR seeker), which makes it possible to implement the "fire-and-forget" homing principle**_


But in Army 3 we have the opportunity to shoot three times and miss the target!
Three useless shots in one target - is that a normal freak!? 

Do you like to shoot at targets in a computer game and see how the missile is exploded out before reaching the target a few dozen meters?  @tbeswick96 if you really think so, then you mentally not healthy or are you a masochist or are you just a liar.

> therefore is not an ACE bug, and therefore does not belong on this issue tracker.

And what prevents the ACE mod try from correcting the BIS error?  Any mod is able to fix a lot in the vanilla game, in addition, ACE mod uses its own system for Titan missiles.

**Comment by ZeusWrath** [comment]
If at least someone would tell me - "yes, you're right - this is not normal,  but we can't fix the error",  then I would say "thank you, sorry" . But instead it, I forced see the denying the existence of an obvious error and a pack of dogs ready to bite me....

LOL, good day

**Comment by synixebrett** [comment]
https://www.youtube.com/watch?v=_LjN3UclYzU

**Comment by jonpas** [comment]
Locking and closing conversation due to inappropriate language towards another contributor. Behave or refrain from posting.
